test/junit/scala/tools/nsc/backend/jvm/opt/InlinerTest.scala

```diff
@@ -1501,16 +1499,48 @@ class InlinerTest extends ClearAfterClass {
   }
 
   @Test
+  def inlineArrayForeachEquivalent(): Unit = {
+    val code =
+      """ trait IndexedSeqOptimized[A] {
+        |   def length: Int
+        |   def apply(i: Int): A
+        |   def foreach[U](f: A => U): Unit = {
+        |    var i = 0
+        |    val len = length
+        |    while (i < len) { f(this(i)); i += 1 }
+        |  }
+        | }
+        | final class ofInt(self: Array[Int]) extends IndexedSeqOptimized[Int] {
+        |    def length = self.length
+        |    def apply(i: Int): Int = self(i)
+        | }
+        |
+        |class C {
+        |  def consume(x: Int) = ()
+        |  // was: a.foreach(consume), but I think this requires the STARR that built the stdlib to be the same as
+        |  // the compiler used to compile this snippet with regards to naming conventions of lambda impl methods.
+        |  def t(a: Array[Int]): Unit = new ofInt(a).foreach(consume)
+        |}
+      """.stripMargin
+    val c = compile(code).find(_.name == "C").get
+    val t = getSingleMethod(c, "t")
+    assertNoIndy(t)
+    assertInvoke(t, "C", "$anonfun$t$1")
+  }
+
+  @Test
+  @Ignore // TODO figure out exactly why this fails
```

Comment:
yes, it will prevent confusions like in this case.. it will work once we have a new starr though :)